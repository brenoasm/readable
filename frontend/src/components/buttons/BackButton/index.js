import React from 'react';
import PropTypes from 'prop-types';

import Button from '../Button/index';

const propTypes = {
  handleClick: PropTypes.func.isRequired
};

const defaultProps = {
  handleClick: () => {}
};

const BackButton = ({ handleClick }) => (
  <Button text="Voltar" handleClick={handleClick} />
);

BackButton.propTypes = propTypes;
BackButton.defaultProps = defaultProps;

export default BackButton;
