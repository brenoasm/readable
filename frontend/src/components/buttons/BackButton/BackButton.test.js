/* eslint-disable */

import React from 'react';

import BackButton from './';

describe('BackButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<BackButton handleClick={jest.fn()}/>);
  })

  afterEach(() => {
    wrapper.unmount();
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
