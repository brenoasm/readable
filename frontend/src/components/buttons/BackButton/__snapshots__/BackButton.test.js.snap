// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackButton
    handleClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "color": "#236A62",
      "disabled": false,
      "handleClick": [Function],
      "text": "Voltar",
      "textColor": "#fff",
      "type": "button",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": "#236A62",
        "disabled": false,
        "handleClick": [Function],
        "text": "Voltar",
        "textColor": "#fff",
        "type": "button",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
